source/io/CsvReader.ooc

```diff
@@ -72,12 +72,16 @@ CsvReader: class extends Iterator<VectorList<Text>> {
 		result
 	}
 	delimiter ::= static ','
-	open: static func (filename: Text) -> This {
+	open: static func ~text (filename: Text) -> This {
+		This open(filename toString())
+	}
+	open: static func ~string (filename: String) -> This {
 		result: This = null
-		file := File new(filename toString())
+		file := File new(filename)
 		if (file exists?())
 			result = This new(FileReader new(file))
 		file free()
+		filename free()
```

Comment:
Or, alternatively, the main code is in the `Text` overload, and all the `String` one does is call the `Text` version.